build/dsc/resource.rb

```diff
@@ -82,7 +82,37 @@ def ensurable?
 
     def absentable?
       properties.detect do |p|
-        p.is_ensure? && p.values.any? { |v| v.casecmp('absent') == 0 }
+        p.is_ensure? && p.values.any? { |v| v.casecmp('absent') == 0 || v.casecmp('disable') == 0 }
+      end
+    end
+
+     def absent_value
+      properties.detect do |p|
+        p.is_ensure? && p.values.each do |v|
+          return 'absent' if v.casecmp('absent') == 0
+          return 'disable' if v.casecmp('disable') == 0
+        end
+      end
+    end
+
+    def ensure_value
+      properties.detect do |p|
+        p.is_ensure? && p.values.each do |v|
+          return 'present' if v.casecmp('present') == 0
+          return 'enable' if v.casecmp('enable') == 0
+        end
+      end
+    end
+
+    def absent_opposite?
+      properties.detect do |p|
+        p.is_ensure? && p.values.any? { |v| v.casecmp('disable') == 0 }
+      end
+    end
+
+    def ensure_opposite?
+      properties.detect do |p|
+        p.is_ensure? && p.values.any? { |v| v.casecmp('enable') == 0 }
```

Comment:
I don't believe we need `ensure_opposite?` (the opposite functions) now that we are returning the `ensure_value` and `absent_value`.